### ISSUE TITLE
Error running vike build

### ISSUE THREAD

**Issue by CannonLock** [vikejs/vike#1359]
### Description

When trying to run `yarn build` on our [website](https://github.com/UW-Macrostrat/web).

```
Error: [vike@0.4.150][Bug] You stumbled upon a bug in Vike's source code. Go to https://github.com/vikejs/vike/issues/new and copy-paste this error; a maintainer will fix the bug (usually under 24 hours).
    at crawlPlusFiles (file:///Users/clock/WebstormProjects/web/.yarn/__virtual__/vike-virtual-6b559751aa/0/cache/vike-npm-0.4.150-b5b0d5e946-3c925b3339.zip/node_modules/vike/dist/esm/node/plugin/plugins/importUserCode/v1-design/getVikeConfig/crawlPlusFiles.js:12:5)
    at findPlusFiles (file:///Users/clock/WebstormProjects/web/.yarn/__virtual__/vike-virtual-6b559751aa/0/cache/vike-npm-0.4.150-b5b0d5e946-3c925b3339.zip/node_modules/vike/dist/esm/node/plugin/plugins/importUserCode/v1-design/getVikeConfig.js:798:25)
    at loadInterfaceFiles (file:///Users/clock/WebstormProjects/web/.yarn/__virtual__/vike-virtual-6b559751aa/0/cache/vike-npm-0.4.150-b5b0d5e946-3c925b3339.zip/node_modules/vike/dist/esm/node/plugin/plugins/importUserCode/v1-design/getVikeConfig.js:73:29)
    at loadVikeConfig (file:///Users/clock/WebstormProjects/web/.yarn/__virtual__/vike-virtual-6b559751aa/0/cache/vike-npm-0.4.150-b5b0d5e946-3c925b3339.zip/node_modules/vike/dist/esm/node/plugin/plugins/importUserCode/v1-design/getVikeConfig.js:233:46)
    at loadVikeConfig_withErrorHandling (file:///Users/clock/WebstormProjects/web/.yarn/__virtual__/vike-virtual-6b559751aa/0/cache/vike-npm-0.4.150-b5b0d5e946-3c925b3339.zip/node_modules/vike/dist/esm/node/plugin/plugins/importUserCode/v1-design/getVikeConfig.js:196:21)
    at getVikeConfig (file:///Users/clock/WebstormProjects/web/.yarn/__virtual__/vike-virtual-6b559751aa/0/cache/vike-npm-0.4.150-b5b0d5e946-3c925b3339.zip/node_modules/vike/dist/esm/node/plugin/plugins/importUserCode/v1-design/getVikeConfig.js:68:29)
    at getConfigVikPromise (file:///Users/clock/WebstormProjects/web/.yarn/__virtual__/vike-virtual-6b559751aa/0/cache/vike-npm-0.4.150-b5b0d5e946-3c925b3339.zip/node_modules/vike/dist/esm/node/plugin/plugins/config/index.js:27:60)
```

**Comment by brillout** [comment]
Thank you for opening the bug report.

Insert the following before the line `12` of `/Users/clock/WebstormProjects/web/.yarn/__virtual__/vike-virtual-6b559751aa/0/cache/vike-npm-0.4.150-b5b0d5e946-3c925b3339.zip/node_modules/vike/dist/esm/node/plugin/plugins/importUserCode/v1-design/getVikeConfig/crawlPlusFiles.js` and report back the log result.

```diff
    // Vike prepends userRootDir without resolving, ...
+   console.log('userRootDir', userRootDir);
+   console.log('outDirAbsoluteFilesystem', outDirAbsoluteFilesystem);
    assert(outDirAbsoluteFilesystem.startsWith(userRootDir));
```

**Comment by CannonLock** [comment]
userRootDir /Users/clock/WebstormProjects/web/src
outDirAbsoluteFilesystem /Users/clock/WebstormProjects/web/dist/
error during build:
Error: [vike@0.4.150][Bug] You stumbled upon a bug in Vike's source code. Go to https://github.com/vikejs/vike/issues/new and copy-paste this error; a maintainer will fix the bug (usually under 24 hours).
    at crawlPlusFiles (file:///Users/clock/WebstormProjects/web/node_modules/vike/dist/esm/node/plugin/plugins/importUserCode/v1-design/getVikeConfig/crawlPlusFiles.js:14:5)
    at findPlusFiles (file:///Users/clock/WebstormProjects/web/node_modules/vike/dist/esm/node/plugin/plugins/importUserCode/v1-design/getVikeConfig.js:798:25)
    at loadInterfaceFiles (file:///Users/clock/WebstormProjects/web/node_modules/vike/dist/esm/node/plugin/plugins/importUserCode/v1-design/getVikeConfig.js:73:29)
    at loadVikeConfig (file:///Users/clock/WebstormProjects/web/node_modules/vike/dist/esm/node/plugin/plugins/importUserCode/v1-design/getVikeConfig.js:233:46)
    at loadVikeConfig_withErrorHandling (file:///Users/clock/WebstormProjects/web/node_modules/vike/dist/esm/node/plugin/plugins/importUserCode/v1-design/getVikeConfig.js:196:21)
    at getVikeConfig (file:///Users/clock/WebstormProjects/web/node_modules/vike/dist/esm/node/plugin/plugins/importUserCode/v1-design/getVikeConfig.js:68:29)
    at getConfigVikPromise (file:///Users/clock/WebstormProjects/web/node_modules/vike/dist/esm/node/plugin/plugins/config/index.js:27:60)

**Comment by nitedani** [comment]
What's inside your src directory? Isn't vite config accidentally in your src directory?

**Comment by davenquinn** [comment]
are you somehow trying to use the "v1 design" for this page / set of pages when we haven't rolled over to that yet? That is a decision that will have to be taken at the whole-application level. It doesn't seem like you would be doing this, so perhaps it's mistakenly inferring that this is the case

**Comment by CannonLock** [comment]
@davenquinn I got these errors building the `main` branch.

**Comment by nitedani** [comment]
`userRootDir /Users/clock/WebstormProjects/web/src` this is wrong
What's in your src directory?

**Comment by brillout** [comment]
Vike doesn't need to (and shouldn't) assume anything in that regard. I removed the assertion; AFAICT it should work just fine.

Pre-released in `0.4.150-commit-63b1c32`. Let us know if you still run into any issue.
